### PR TITLE
[MOB-5737] schedule a default timer of 10s if auth token null

### PR DESCRIPTION
**Pull request by roninopf**
## 🔹 Jira Ticket(s)

* [MOB-5737](https://iterable.atlassian.net/browse/MOB-5737)

## ✏️ Description

> Please provide a brief description of what this pull request does.

[MOB-5737]: https://iterable.atlassian.net/browse/MOB-5737?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by evantk91 — APPROVED** [review]
LGTM 👍

File: swift-sdk/Internal/AuthManager.swift [hunk 1]
```diff
@@ -107,14 +107,22 @@ class AuthManager: IterableAuthManagerProtocol {
     private func queueAuthTokenExpirationRefresh(_ authToken: String?) {
         ITBInfo()
         
+        clearRefreshTimer()
+        
         guard let authToken = authToken, let expirationDate = AuthManager.decodeExpirationDateFromAuthToken(authToken) else {
+            /// schedule a default timer of 10 seconds if we fall into this case
+            scheduleAuthTokenRefreshTimer(10)
```
Comment (Ayyanchira) [inline swift-sdk/Internal/AuthManager.swift:114]: Liked the guard let else use 👍🏼

File: swift-sdk/Internal/AuthManager.swift [hunk 2]
```diff
@@ -107,14 +107,22 @@ class AuthManager: IterableAuthManagerProtocol {
...
         }
         
         let timeIntervalToRefresh = TimeInterval(expirationDate) - dateProvider.currentDate.timeIntervalSince1970 - expirationRefreshPeriod
         
-        clearRefreshTimer()
-        expirationRefreshTimer = Timer.scheduledTimer(withTimeInterval: timeIntervalToRefresh, repeats: false) { [weak self] _ in
+        scheduleAuthTokenRefreshTimer(timeIntervalToRefresh)
+    }
+    
+    private func scheduleAuthTokenRefreshTimer(_ interval: TimeInterval) {
```
Comment (Ayyanchira) [inline swift-sdk/Internal/AuthManager.swift:124]: Method extracted looks helpful 👍🏼. We might have to call it from another few places too.

**Review by Ayyanchira — APPROVED** [review]
LGTM 👍🏼

**Review by davidtruong — APPROVED** [review]
LGTM. We'll want to create a separate ticket to handle the failure callbacks when the retrievedAuthToken is null or fails to decode.

**Pull request merged**